File: workflow-payment-java/src/main/java/com/acme/workflows/payment/java/Worker.java
package com.acme.workflows.payment.java;

import io.infinitic.workers.InfiniticWorker;

public class Worker {
    public static void main(String[] args) {
        try (InfiniticWorker worker = InfiniticWorker.fromConfigResource("/infinitic.yml", "/payment.yml")) {
            worker.start();
        }
    }
}

File: workflow-payment-kotlin/src/main/kotlin/com/acme/workflows/payment/kotlin/Worker.kt
package com.acme.workflows.payment.kotlin

import io.infinitic.workers.InfiniticWorker

fun main() {
    // infinitic.yml resource is from the dependant "contracts" module
    InfiniticWorker.fromConfigResource("/infinitic.yml", "/payment.yml").use { worker ->
        worker.start()
    }
}
